sdk/tables/azure-data-tables/src/samples/java/TableServiceAsyncClientCodeSnippets.java

```diff
@@ -0,0 +1,199 @@
+            Void -> {},
+            error -> logger.error("There was an error creating the table. Error: " + error),
+            () -> logger.info("Table creation successful."));
+    }
+
+    /**
+     * delete table code snippet
+     */
+    public void deleteTable() {
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        tableServiceAsyncClient.deleteTable("OfficeSupplies").subscribe(
+            Void -> {},
+            error -> logger.error("There was an error deleting the table. Error: " + error),
+            () -> logger.info("Table deletion successful."));
+    }
+
+    /**
+     * query tables code snippet
+     */
+    public void queryTable() {
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+        QueryOptions queryOptions = new QueryOptions();
+        queryOptions.setFilter("TableName eq OfficeSupplies");
+
+        tableServiceAsyncClient.queryTables(queryOptions).subscribe(azureTable -> {
+            logger.info(azureTable.getName());
+        }, error -> {
+            logger.error("There was an error querying the service. Error: " + error);
+        });
+    }
+
+    /**
+     * insert entity code snippet
+     */
+    private void insertEntity() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getTableAsyncClient("OfficeSupplies");
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("RowKey", "crayolaMarkers");
+        properties.put("PartitionKey", "markers");
+
+        tableAsyncClient.createEntity(properties).subscribe(tableEntity -> {
+            logger.info("Insert Entity Successful. Entity: " + tableEntity);
+        }, error -> {
+            logger.error("There was an error inserting the Entity. Error: " + error);
+        });
+    }
+
+    /**
+     * delete entity code snippet
+     */
+    private void deleteEntity() {
+
+        // Build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceClientBuilder()
+            .connectionString("connectionString")
+            .buildAsyncClient();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getTableAsyncClient("OfficeSupplies");
+        String rowKey = "crayolaMarkers";
+        String partitionKey = "markers";
+
+        tableAsyncClient.get(rowKey, partitionKey).flatMap(tableEntity -> {
+            logger.info("Table Entity: " + tableEntity);
+            Mono<Void> deleteEntityMono = tableAsyncClient.deleteEntity(tableEntity);
+            return deleteEntityMono;
```

Comment:
No need for the intermediate local variable `deleteEntityMono`. Just return tableAsyncClient.delete....